src/apps/HeyGPT/HeyGPT/ViewModels/LetsChatViewModel.cs

```diff
@@ -195,5 +280,133 @@ private void AddErrorMessageToHistory(string errorMessage)
         });
     }
 
+    private async Task SendMessageInternal(string userMessage, ChatCharacterDetails? desiredCharacterDetails = null)
+    {
+        AddUserMessageToHistory(userMessage);
+
+        try
+        {
+            var characterDetails = desiredCharacterDetails ?? _characterService.GetSelectedCharacterDetails();
+
+            var extraContext = await GetClipboardContext().ConfigureAwait(true);
+
+            ShouldIncludeClipboardContext = false; // Reset the extra context option after it is used so the user is forced to do it each time.
+
+            var response = await _openAIService.SendPrompt(characterDetails, userMessage, extraContext.Text).ConfigureAwait(true);
+
+            var message = $"{response.Content} [{response.MetaData.Usage.TotalTokens}]";
+
+            AddRobotMessageToHistory(response.CharacterType, message);
+        }
+        catch (Exception e)
+        {
+            AddErrorMessageToHistory(e.ToString());
+        }
+    }
+
+    public static (string Character, string Message) ParseHeyProtocolUri(Uri uri)
```

Comment:
This doesn't really belong here, we should really parse this in the activation handler and deconstruct all the information into a model.  Then the protocol activation handler will just send that information.   We'll do this soon.